satpy/readers/landsat_base.py

```diff
@@ -0,0 +1,815 @@
+        return float(self.root.find(".//IMAGE_ATTRIBUTES/CLOUD_COVER").text)
+
+    def _get_satflag(self, band):
+        """Return saturation flag for a band."""
+        flag = self.root.find(f".//IMAGE_ATTRIBUTES/SATURATION_BAND_{band}").text
+        if flag == "Y":
+            return True
+        return False
+
+    @property
+    def band_saturation(self):
+        """Return per-band saturation flag."""
+        raise NotImplementedError()
+
+    @property
+    def band_calibration(self):
+        """Return per-band saturation flag."""
+        raise NotImplementedError()
+
+    def earth_sun_distance(self):
+        """Return Earth-Sun distance."""
+        return float(self.root.find(".//IMAGE_ATTRIBUTES/EARTH_SUN_DISTANCE").text)
+
+    def build_area_def(self, bname):
+        """Build area definition from metadata."""
+        from pyresample.geometry import AreaDefinition
+
+        # Here we assume that the thermal bands have the same resolution as the reflective bands,
+        # with only the panchromatic band (b08) having a different resolution.
+        if bname in PAN_BANDLIST and self.platform_name in ["Landsat-7", "Landsat-8", "Landsat-9"]:
+            pixoff = float(self.root.find(".//PROJECTION_ATTRIBUTES/GRID_CELL_SIZE_PANCHROMATIC").text) / 2.
+            x_size = float(self.root.find(".//PROJECTION_ATTRIBUTES/PANCHROMATIC_SAMPLES").text)
+            y_size = float(self.root.find(".//PROJECTION_ATTRIBUTES/PANCHROMATIC_LINES").text)
+        else:
+            pixoff = float(self.root.find(".//PROJECTION_ATTRIBUTES/GRID_CELL_SIZE_REFLECTIVE").text) / 2.
+            x_size = float(self.root.find(".//PROJECTION_ATTRIBUTES/REFLECTIVE_SAMPLES").text)
+            y_size = float(self.root.find(".//PROJECTION_ATTRIBUTES/REFLECTIVE_LINES").text)
+
+        # Get remaining geoinfo from file
+        datum = self.root.find(".//PROJECTION_ATTRIBUTES/DATUM").text
+        utm_zone = int(self.root.find(".//PROJECTION_ATTRIBUTES/UTM_ZONE").text)
+        utm_str = f"{utm_zone}N"
+
+        # We need to subtract / add half a pixel from the corner to get the correct extent (pixel centers)
+        ext_p1 = float(self.root.find(".//PROJECTION_ATTRIBUTES/CORNER_UL_PROJECTION_X_PRODUCT").text) - pixoff
+        ext_p2 = float(self.root.find(".//PROJECTION_ATTRIBUTES/CORNER_LR_PROJECTION_Y_PRODUCT").text) - pixoff
+        ext_p3 = float(self.root.find(".//PROJECTION_ATTRIBUTES/CORNER_LR_PROJECTION_X_PRODUCT").text) + pixoff
+        ext_p4 = float(self.root.find(".//PROJECTION_ATTRIBUTES/CORNER_UL_PROJECTION_Y_PRODUCT").text) + pixoff
+
+        # Create area definition
+        pcs_id = f"{datum} / UTM zone {utm_str}"
+        proj4_dict = {"proj": "utm", "zone": utm_zone, "datum": datum, "units": "m", "no_defs": None, "type": "crs"}
+        area_extent = (ext_p1, ext_p2, ext_p3, ext_p4)
+
+        # Return the area extent
+        return AreaDefinition("geotiff_area", pcs_id, pcs_id, proj4_dict, x_size, y_size, area_extent)
```

Comment:
I'm wondering if there a more useful name for the area definition here? Like naming it after the UTM maybe? Like "UTM{utm_zone}"? Or is that unhelpful?

Reply:
I am not sure if this name is visible to user, but giving a more meaningful name definitely would be a good addition. Added it

Reply:
Hm it is visible via `scn[product_name].attrs["area"].area_id`. The "EPSG: XXXX" name may not be super filename friendly. This is a selfish request on my part because I have software that puts the Area ID into the default filename for geotiffs.